yesod-core/src/Yesod/Routes/Class.hs

```diff
@@ -16,10 +20,49 @@ class Eq (Route a) => RenderRoute a where
     renderRoute :: Route a
                 -> ([Text], [(Text, Text)]) -- ^ The path of the URL split on forward slashes, and a list of query parameters with their associated value.
 
+-- | This class acts as a delegation class for 'RenderRoute' on nested
+-- route fragments.
+--
+-- @since 1.6.28.0
+class Eq a => RenderRouteNested a where
+    -- | Render the fragment of the route. To form a complete route, you'll
+    -- need to `mappend` this with the result from the parent `renderRoute`
+    -- or `renderRouteNested` call.
+    --
+    -- @since 1.6.28.0
```

Comment:
I think this requirement is questionable. The instance *can* know about the entire path, and so perhaps we should bake this into the implementation. Would probably make testing subroutes easier.

Reply:
Currently redesigning this, actually - `renderRouteNested` will need to get the parent dyns but will otherwise create the entire route itself.